huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go

```diff
@@ -0,0 +1,406 @@
+)
+
+func DataSourceVpnConnections() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: datasourceConnectionRead,
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"connection_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"gateway_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"gateway_ip": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"vpn_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"connections": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"gateway_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"gateway_ip": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"vpn_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"customer_gateway_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"peer_subnets": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"tunnel_local_address": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tunnel_peer_address": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"enable_nqa": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"ikepolicy": {
+							Type:     schema.TypeList,
+							Elem:     ConnectionIkePolicySchema(),
+							Computed: true,
+						},
+						"ipsecpolicy": {
+							Type:     schema.TypeList,
+							Elem:     ConnectionIpsecPolicySchema(),
+							Computed: true,
+						},
+						"policy_rules": {
+							Type:     schema.TypeList,
+							Elem:     ConnectionsPolicyRuleSchema(),
+							Computed: true,
+						},
+						"created_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"updated_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"enterprise_project_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"connection_monitor_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"ha_role": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func ConnectionIkePolicySchema() *schema.Resource {
```

Comment:
use lowercase will be fine, please modify others.

```suggestion
func connectionIkePolicySchema() *schema.Resource {
```

Reply:
down